DiscordBot/Modules/TimeModule.cs

```diff
@@ -0,0 +1,99 @@
+            }
+            public Geo geo;
+            public string timezone;
+            public decimal timezone_offset;
+            public string date;
+            public string date_time;
+            public string date_time_txt;
+            public string date_time_wti;
+            public string date_time_ymd;
+            public string date_time_unix;
+            public string time_24;
+            public string time_12;
+            public uint week;
+            public uint month;
+            public uint year;
+            public string year_abbr;
+            public bool is_dst;
+            public int dst_savings;
+        }
+#pragma warning restore 0649
+
+        private List<string> _commandList = new List<string>();
+
+        private Settings.Deserialized.Settings _settings;
+
+        public TimeModule(Settings.Deserialized.Settings settings, CommandHandlingService commandHandlingService)
+        {
+            _settings = settings;
+
+            Task.Run(async () =>
+            {
+                var commands = await commandHandlingService.GetCommandList("TimeModule", true, true, false);
+                _commandList = commands.MessageSplitToSize();
+            });
+        }
+
+        [Command("time"), Alias("timezone"), Summary("Find the locale time of a location.")]
+        public async Task Time(params string[] location)
+        {
+            try
+            {
+                using (var wc = new WebClient())
+                {
+                    var res = await wc.DownloadStringTaskAsync($"https://api.ipgeolocation.io/timezone?apiKey={_settings.IPGeolocationAPIKey}&location={string.Join(" ", location)}");
+                    var response = JsonConvert.DeserializeObject<Response>(res);
+
+                    var embedBuilder = new EmbedBuilder();
+
+                    List<string> geo = new List<string>();
+                    if (!String.IsNullOrEmpty(response.geo.city)) geo.Add(response.geo.city);
+                    if (!String.IsNullOrEmpty(response.geo.state)) geo.Add(response.geo.state);
+                    if (!String.IsNullOrEmpty(response.geo.country)) geo.Add(response.geo.country);
+                    embedBuilder.WithTitle($"Time in {String.Join(", ", geo)}");
+
+                    embedBuilder.WithDescription($"{response.date_time_txt}");
+                    embedBuilder.AddField("Timezone", $"{response.timezone} ({response.timezone_offset:+#.#;-#.#;+0})", true);
+
+                    var dst = $"{(response.is_dst ? "Yes" : "No")}";
+                    if (response.is_dst) dst += $" ({response.dst_savings:+#.#;-#.#;+0})";
+                    embedBuilder.AddField("DST", dst, true);
+
+                    var embed = embedBuilder.Build();
+                    await ReplyAsync(embed: embed);
+                }
+            }
+            catch (Exception e)
+            {
+                Console.Error.WriteLine(e);
+                await ReplyAsync($"{e.Message}");
```

Comment:
Do we want to report this straight into chat? 

This module has settings pulled into it, might be better to report the full message to bot-announcements and a generic "something broke" in chat.

Reply:
Yeah true